src/containers/Elements/NodeForm.js

```diff
@@ -51,26 +74,51 @@ class NodeForm extends Component {
     }, this.submit);
   };
 
+
+  close = () => {
+    this.props.closeModal(this.props.name);
+    this.setState({
+      fieldIndex: 0,
+    });
+  }
+
   render() {
     const {
       title,
       fields,
+      autoPopulate,
       name,
       addAnother,
       initialValues,
     } = this.props;
 
+    const large = window.matchMedia('screen and (min-device-aspect-ratio: 16/9)').matches;
+
+    const previousElement = (
+      <div>
+        {this.state.fieldIndex !== 0 && <Icon name="form-arrow-left" onClick={this.previousField} />}
+      </div>);
+    const lastField = this.state.fieldIndex === fields.length - 1;
+    const nextElement = lastField || large ? null :
+      (<Icon name="form-arrow-right" />);
+
     return (
-      <Modal name={name} title={title}>
+      <Modal name={name} title={title} close={this.close} className={large ? '' : 'modal--mobile'}>
+        { !large && <div className="modal__pips">
+          <Pips count={fields.length} currentIndex={this.state.fieldIndex} />
+        </div>}
         <Form
-          fields={fields}
-          initialValues={!addAnother ? initialValues : null}
+          fields={large ? fields : [fields[this.state.fieldIndex]]}
+          autoPopulate={autoPopulate}
+          initialValues={initialValues}
           autoFocus
           form={name.toString()}
-          onSubmit={this.onSubmit}
+          onSubmit={lastField || large ? this.onSubmit : this.nextField}
```

Comment:
What does this do? Can we make it easier to read with named functions / variables?

Reply:
It decides if we need to submit, or if we need to go to the next field (depending on if we are large). Seems pretty readable to me.  I guess it could be broken out into a function. 

How do we feel about inline comments for things like this? Should the code be self-explanatory?

Reply:
I read it as: "string | probablyBoolean ? eventHandler() : propertyGetterProbablyString()"

So if it's like you say I'd change it to:

`onSubmit`

Reply:
Makes sense. I can make these changes!